### PR TITLE
Add feature for issue#1809

### PR DESCRIPTION
**Pull request by tigercl**

### CODE REVIEW

File: etc/emqx.conf [hunk 1]
```diff
@@ -1346,6 +1351,11 @@ listener.ws.external.nodelay = true
 ## Examples: 8084, 127.0.0.1:8084, ::1:8084
 listener.wss.external = 8084
 
+## Whether the client must include "mqtt" in the list of WebSocket Sub Protocols it offers
+##
+## Value: true | false
+listener.wss.external.standard_mqtt = true
```
Comment (emqplus) [inline etc/emqx.conf]: listener.ws.external.mqtt_path = /mqtt

File: priv/emqx.schema [hunk 2]
```diff
@@ -1094,6 +1094,11 @@ end}.
   {datatype, [integer, ip]}
 ]}.
 
+{mapping, "listener.ws.$name.standard_mqtt", "emqx.listeners", [
+  {default, true},
+  {datatype, {enum, [true, false]}}
+]}.
```
Comment (emqplus) [inline priv/emqx.schema]:
```
{mapping, "listener.ws.$name.mqtt_path", "emqx.listeners", [
   {default, "/mqtt"},
   {datatype, string}
 ]}.
```

File: priv/emqx.schema [hunk 3]
```diff
@@ -1195,6 +1200,11 @@ end}.
   {datatype, [integer, ip]}
 ]}.
 
+{mapping, "listener.wss.$name.standard_mqtt", "emqx.listeners", [
+  {default, true},
+  {datatype, {enum, [true, false]}}
+]}.
+
```
Comment (emqplus) [inline priv/emqx.schema]:
```
{mapping, "listener.wss.$name.mqtt_path", "emqx.listeners", [
   {default, "/mqtt"},
   {datatype, string}
 ]}.
```

File: priv/emqx.schema [hunk 4]
```diff
@@ -1365,7 +1375,8 @@ end}.
                 end,
 
     LisOpts = fun(Prefix) ->
-                  Filter([{acceptors,   cuttlefish:conf_get(Prefix ++ ".acceptors", Conf)},
+                  Filter([{standard_mqtt, cuttlefish:conf_get(Prefix ++ ".standard_mqtt", Conf, undefined)},
```
Comment (emqplus) [inline priv/emqx.schema]:
```
{mqtt_path, cuttlefish:conf_get(Prefix ++ ".mqtt_path", Conf, undefined)},
```

File: src/emqx_listeners.erl [hunk 5]
```diff
@@ -51,12 +51,12 @@ start_listener(Proto, ListenOn, Options) when Proto == ssl; Proto == tls ->
 
 %% Start MQTT/WS listener
 start_listener(Proto, ListenOn, Options) when Proto == http; Proto == ws ->
-    Dispatch = cowboy_router:compile([{'_', [{"/mqtt", emqx_ws_connection, Options}]}]),
+    Dispatch = cowboy_router:compile([{'_', [{subprotocol_name(Options), emqx_ws_connection, Options}]}]),
```
Comment (emqplus) [inline src/emqx_listeners.erl]:
```
mqtt_path(Options)
```

File: src/emqx_listeners.erl [hunk 6]
```diff
@@ -51,12 +51,12 @@ start_listener(Proto, ListenOn, Options) when Proto == ssl; Proto == tls ->
...
     start_http_listener(fun cowboy:start_clear/3, 'mqtt:ws', ListenOn, ranch_opts(Options), Dispatch);
 
 %% Start MQTT/WSS listener
 start_listener(Proto, ListenOn, Options) when Proto == https; Proto == wss ->
-    Dispatch = cowboy_router:compile([{'_', [{"/mqtt",  emqx_ws_connection, Options}]}]),
+    Dispatch = cowboy_router:compile([{'_', [{subprotocol_name(Options),  emqx_ws_connection, Options}]}]),
```
Comment (emqplus) [inline src/emqx_listeners.erl]: mqtt_path(Options)

File: src/emqx_listeners.erl [hunk 7]
```diff
@@ -67,6 +67,12 @@ start_mqtt_listener(Name, ListenOn, Options) ->
 start_http_listener(Start, Name, ListenOn, RanchOpts, Dispatch) ->
     Start(Name, with_port(ListenOn, RanchOpts), #{env => #{dispatch => Dispatch}}).
 
+subprotocol_name(Options) ->
+    case proplists:get_value(standard_mqtt, Options, true) of
+        true -> "/mqtt";
+        false -> "/"
+    end.
+
```
Comment (emqplus) [inline src/emqx_listeners.erl]:
```
mqtt_path(Options) ->
    proplists:get_value(mqtt_path, Options, "/mqtt").
```

File: src/emqx_protocol.erl [hunk 8]
```diff
@@ -249,6 +249,13 @@ preprocess_properties(Packet = #mqtt_packet{
                       PState = #pstate{proto_ver = ?MQTT_PROTO_V5, topic_aliases = Aliases}) ->
     {Packet, PState#pstate{topic_aliases = maps:put(AliasId, Topic, Aliases)}};
 
+preprocess_properties(Packet = #mqtt_packet{variable = #mqtt_packet_publish{properties = #{'Message-Expiry-Interval' := _Interval}}},
+                      PState = #pstate{proto_ver = ?MQTT_PROTO_V5}) ->
+    {Packet, PState};
+preprocess_properties(Packet = #mqtt_packet{variable = Publish = #mqtt_packet_publish{properties = Properties}},
+                      PState = #pstate{proto_ver = ?MQTT_PROTO_V5}) ->
+    {Packet#mqtt_packet{variable = Publish#mqtt_packet_publish{properties = maps:put('Message-Expiry-Interval', 0, Properties)}}, PState};
+
```
Comment (emqplus) [inline src/emqx_protocol.erl]: Why put a default 'Message-Expiry-Interval'???

Comment (tigercl) [reply src/emqx_protocol.erl]: > Why put a default 'Message-Expiry-Interval'???

These code used in retainer plugin, I push it together with code of issue#1809 unexpectedly.

**Pull request merged**